### PR TITLE
Add naming conventions for mm tables

### PR DESCRIPTION
**Pull request by sypets**
The naming conventions page already contains several rules
and examples for database tables.

A section is now added for mm tables which contain records
for multiple-multiple relations between tables.

The rules for Extbase are clear.

For non-Extbase no such rules currently exist, but it is
suggested to recommend a similar rule as in the Extbase
tables.

### CODE REVIEW

File: Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst [hunk 1]
```diff
@@ -220,6 +220,27 @@ Examples for Extbase domain models and table names of an extension named `cool_s
 | :php:`Vendor\CoolShop\Domain\Model\Billing\Address` | :sql:`tx_coolshop_domain_model_billing_address` |
 +-----------------------------------------------------+-------------------------------------------------+
 
+MM tables (for multiple-multiple relations between tables) follow these rules.
+
+Extbase tables SHOULD follow this pattern:
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst:225]: Some lines above "should" is in lowercase and bold  ("Database table names should follow this pattern"), we should stick to one formatting.

File: Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst [hunk 2]
```diff
@@ -220,6 +220,27 @@ Examples for Extbase domain models and table names of an extension named `cool_s
...
+
+.. code-block:: none
+
+   # rule for Extbase
+   tx_<extension-prefix>_domain_model_<model-name-1>_<model-name-2>_mm
+   # example: EXT:news with relation between news and tag
+   tx_news_domain_model_news_tag_mm
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst:232]: Either use the above blogexample post scenario for consistency (relation between post and tag) or move the tip below before the new MM tables section directly under the cool_shop example.

File: Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst [hunk 3]
```diff
@@ -220,6 +220,27 @@ Examples for Extbase domain models and table names of an extension named `cool_s
...
+
+   # recommendation for non-Extbase third party extensions
+   tx_<extension-prefix>_<model-1>_<model-2>_mm
+
+   # example for TYPO3 core:
+   sys_category_record_mm
+
 .. tip::
 
    You may notice, that the names above use the singular form, e.g. `post` and
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst:246]: When you are on it, please adjust the indentation here.

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst (hunk 2 above)
Comment (sypets) [reply Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst:232]: ok, will push another commit.

File: Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst (hunk 1 above)
Comment (sypets) [reply Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst:225]: The SHOULD (in capital letters) is explained above "Hard requirements are emphasized by using the words MUST, etc. as specified in https://www.ietf.org/rfc/rfc2119.txt", but you are right, that is currently not consistently followed. I will change this to conform to the used conventions.

File: Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst [hunk 4]
```diff
@@ -221,11 +221,34 @@ Examples for Extbase domain models and table names of an extension named `cool_s
...
+**MM** tables (for multiple-multiple relations between tables) follow these rules.
+
+Extbase:
+
+.. code-block:: none
+
+   # rule for Extbase
+   tx_<extension-prefix>_domain_model_<model-name-1>_<model-name-2>_mm
+   # example: EXT:news with relation between news and tag
+   tx_blogexample_domain_model_post_comment
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst:238]: trailing `_mm` is missing

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst (hunk 4 above)
Comment (sypets) [reply Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst]: Thank you. Is fixed.

**Review by brotkrueml — APPROVED** [review]

**Review by tomwarwick — APPROVED** [review]

**Pull request merged**